### ISSUE TITLE
Add setting to disable token rotation animation?

### ISSUE THREAD

**Issue by GambetTV** [Varriount/fvtt-autorotate#26]
It would be great if you could add a setting to disable token's animating their rotation (either by holding Shift + Mousewheel, Ctrl + Mousewheel, Shift + WASD, or using your module and moving the token in different directions). It causes all kinds of visual problems in game. The module About Face added this setting and it's really helped, but unfortunately About Face has a lot of bloat with their direction indicators that causes all kinds of problems with other modules. Autorotate is a lot leaner of a module, and would be my preferred choice if moving tokens didn't look so janky because of the rotation animation.

**Comment by GambetTV** [comment]
[2022-12-07 23-27-22.webm](https://user-images.githubusercontent.com/72000497/206360558-1df59cf4-973d-4dbf-adb8-0e5196121f72.webm)

**Comment by Varriount** [comment]
Ok, so it looks like what About Face does is [patch the internal animation](https://github.com/mclemente/about-face/blob/1083ac8d517b289392ee79c91dbb049635f15b7c/scripts/logic.js#L21) procedure:
```javascript
toggleAnimateFrame(value) {
    if (value) libWrapper.register(MODULE_ID, "CanvasAnimation._animateFrame", _animateFrame, "OVERRIDE");
    else libWrapper.unregister(MODULE_ID, "CanvasAnimation._animateFrame");
}
```
```javascript
function _animateFrame(deltaTime, animation) {
	const { attributes, duration, ontick } = animation;

	// Compute animation timing and progress
	const dt = this.ticker.elapsedMS; // Delta time in MS
	animation.time += dt; // Total time which has elapsed
	const pt = animation.time / duration; // Proportion of total duration
	const complete = animation.time >= duration;
	const pa = complete ? 1 : animation.easing ? animation.easing(pt) : pt;

	// Update each attribute
	try {
		for (let a of attributes) {
			// Snap to final target
			if (complete) {
				a.parent[a.attribute] = a.to;
				a.done = a.delta;
			}

			// Continue animating
			else {
				const da =
					a.delta *
					((a.attribute == "rotation" && [2, 3].includes(game.aboutFace.disableAnimations)) ||
					((a.attribute == "scaleX" || a.attribute == "scaleY") && [1, 3].includes(game.aboutFace.disableAnimations))
						? 1
						: pa);
				a.parent[a.attribute] = a.from + da;
				a.done = da;
			}
		}

		// Callback function
		if (ontick) ontick(dt, animation);
	} catch (err) {
		// Terminate the animation if any errors occur
		animation.reject(err);
	}

	// Resolve the original promise once the animation is complete
	if (complete) animation.resolve(true);
}
```

I would prefer not to do this, as it's really something that should probably be fixed in Foundry itself - the current behavior is arguably a bug.

There *might* be a way to do this without patching Foundry by separating the rotation and movement changes, but I would have to test first.

**Comment by Shuggaloaf** [comment]
I noticed this same "animation" behavior. So if I'm understanding correctly this is a core feature and has nothing to do with this mod?

**Comment by Varriount** [comment]
> I noticed this same "animation" behavior. So if I'm understanding correctly this is a core feature and has nothing to do with this mod?

@Shuggaloaf Correct. When a token's position (including rotation) is changed, Foundry does a "linear" animated transition from the old position/rotation to the new one. Foundry's current API doesn't provide any way to customize this animation, so a module developer's only solution is to patch its implementation.

**Comment by Shuggaloaf** [comment]
@Varriount Thanks for the reply. 

It's too bad that the animation plays out over the whole move when you drag it long distances. The animation actually looks good when you're just moving from square to square. 

If it could turn first, then move I think that would be ideal. I'm guessing that's probably something that can't be done within the module though? 

Or maybe a better solution is do not animate if being moved more than 1 square/step but allow animation if only 1.
